torchhd/datasets/abalone.py

```diff
@@ -0,0 +1,165 @@
+import os
+import os.path
+from typing import Callable, Optional, Tuple, List
+import torch
+from torch.utils import data
+import pandas as pd
+import tarfile
+import numpy as np
+
+from .utils import download_file
+
+
+class Abalone(data.Dataset):
+    """`Abalone <https://archive.ics.uci.edu/ml/datasets/abalone>`_ dataset.
+
+    Args:
+        root (string): Root directory containing the files of the dataset.
+        train (bool, optional): If True, returns training (sub)set stored in ``train.data`` as further determined by fold_id and fold_train variables.
```

Comment:
It seems that train has to be set to true otherwise an error is thrown in load data so I'm not sure why the argument is there?

Reply:
Good point, @mikeheddes . I was mainly thinking in terms of being general for all the datasets in this collection and also with the datasets you already have in the library. 
The thing about "abalone" is that it does not have a separate file for test set and, therefore, the idea is to issue error. Some other datasets, however, e.g., "annealing" have separate files for training and testing. So for this dataset setting train to false will return data in "annealing_test_R.dat" file. I was trying to briefly make this point in the comment but easily admit that it is confusing. Do you think it makes more sense now after the explanation?

Reply:
Sorry, I think it is actually more complicated than that. It looks like I confused two types of folds. There is one stored in "conxuntos.dat" and the other one in "conxuntos_kfold.dat". I probably want to chat with you @mikeheddes to discuss what is the best way to tackle it.

Reply:
@mikeheddes, I was thinking about it for a while and think that I have better structure in the new code (thanks to many of your suggestions!).